### ISSUE TITLE
Compiler should throw error on non-root query node with @refetchable directive

### ISSUE THREAD

**Issue by tinleym** [facebook/relay#4059]
For the following schema and query

```
type Query {
  authz: Authz
  timeline(
    first: Int
    last: Int
    before: Cursor
    after: Cursor
  ): TimelineConnection!
}

union Authz = RegisteredAuthz | GuestAuthz

type RegisteredAuthz {
  query: Query!
}

type GuestAuthz  {
  query: Query!
}

[TimelineConnection types]
```

```
fragment GuestTimeline_Query on Query
@refetchable(queryName: "GuestTimelineQuery")
@argumentDefinitions(
  first: { type: "Int", defaultValue: 20 }
  after: { type: "Cursor" }
 ) {
  timeline(
    first: $first
    after: $after
  )
    @connection(key: "GuestTimeline__timeline") {
    [TimelineConnection query]
  }
}
```

Calling loadNext from usePaginationFragment on GuestTimeline_Query fetches on the root, even if GuestTimeline_Query was not initially called on the root (and instead called on GuestAuthz). This prevents the connection from updating.

Since this usage is all but guaranteed to yield an unintended result, I think it would be helpful for the compiler to reject it.

Note this approach does work (updates the connection) if the refetchable fragment is called on a non-root query parent (like GuestAuthz), provided said parent implements the Node interface.

**Comment by tinleym** [comment]
This is a duplicate of #4015
